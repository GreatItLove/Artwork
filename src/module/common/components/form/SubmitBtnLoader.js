import React from 'react';
import RefreshIndicator from 'material-ui/RefreshIndicator';
import RaisedButton from 'material-ui/RaisedButton';
import PropTypes from 'prop-types';

const propTypes = {
  label: PropTypes.string.isRequired,
  loading: PropTypes.bool.isRequired
};

/**
 * @function SubmitBtnLoader
 * @param {String} label
 * @param {Boolean} loading
 * @param {Object} rest
 * @returns {XML}
 */
function SubmitBtnLoader({ label, loading, ...rest }) {
  const { dispatch, ...restProps } = rest;
  return loading ? (
    <RefreshIndicator style={{ position: 'relative', margin: '0 auto' }} left={0} top={0} status="loading" />
  ) : (
      <RaisedButton type="submit" label={label} primary={true} fullWidth={true} {...restProps} />
    );
}

SubmitBtnLoader.propTypes = propTypes;

export default SubmitBtnLoader;
