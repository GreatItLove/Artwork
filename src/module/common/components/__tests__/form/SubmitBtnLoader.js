import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import SubmitBtnLoader from '../../form/SubmitBtnLoader';

describe('SubmitBtnLoader', () => {
  it('SubmitBtnLoader snapshot btn', () => {
    const wrapper = shallow(<SubmitBtnLoader label="some label" loading={false} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('SubmitBtnLoader snapshot loading', () => {
    const wrapper = shallow(<SubmitBtnLoader label="some label" loading={true} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
