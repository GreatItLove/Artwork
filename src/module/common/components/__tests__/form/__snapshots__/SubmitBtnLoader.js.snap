// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitBtnLoader SubmitBtnLoader snapshot btn 1`] = `
<RaisedButton
  disabled={false}
  fullWidth={true}
  label="some label"
  labelPosition="after"
  primary={true}
  secondary={false}
  type="submit"
/>
`;

exports[`SubmitBtnLoader SubmitBtnLoader snapshot loading 1`] = `
<RefreshIndicator
  left={0}
  percentage={0}
  size={40}
  status="loading"
  style={
    Object {
      "margin": "0 auto",
      "position": "relative",
    }
  }
  top={0}
/>
`;
